### DISCUSSION TITLE
Visual Studio Code with "Modern FORTRAN

### DISCUSSION THREAD

**Discussion by jbaben** [krvajal/vscode-fortran-support#1008] [Q&A - deprecated see Fortran Discourse]
I added the “Modern Fortran” extension to Visual Studio Code (version: 1.84.2) and added the corresponding path to “fortls”. 
When I now want to compile the FORTRAN-Code under “Run and debug”, I get the message: “Linter can't be found in $PATH”. 
What does that mean, I actually made all the settings for “Modern Fortran” (Path to: Compiler, fortls, fprettify).
The problem is: unfortunately I have no idea what “Linter” is supposed to be.
I hope you can help me.
Kind Regards
Juergen

**Comment by gnikit** [comment]
Hi @jbaben, can you please first install the pre-release of the extension. 
A [linter](https://en.wikipedia.org/wiki/Lint_(software)) is a piece of software that helps you detect errors. In Fortran we use the actual compilers e.g. GFortran or Intel compilers to detect certain errors. So you need to install those first.

If you could post the JSON settings file to get an idea of what is going on and a list of the installed vscode extensions that would also help.

**Comment by jbaben** [comment]
Hello,
thank you for your fast answer.
For the first I have install the pre-release of "Modern Fortran" Version: 3.4.2023110901
I have download the Fortran-Compiler from: http://www.equation.com/servlet/equation.cmd?fa=fortran#google_vignette
File: gcc-13.2.0-64.exe -> gfortran
My "settings.json":
{
    "fortran.fortls.path": "C:\\Users\\jbabe\\AppData\\Roaming\\Python\\Python310\\",
    "fortran.linter.compilerPath": "C:\\Strawbwrry\\Fortran\\bin\\"
}
extensions.json:
{
    "recommendations": [
        "fortran-lang.linter-gfortran"
    ]
}
I have a lot of extensions in my vscode: can I save the installed extensions to a file?
The following extensions are shown as deprecated:
Node.js Extension Pack
npm
TypeScript Hero

Kind Regards
Juergen

**Reply by gnikit** [reply]
> "fortran.fortls.path": "C:\Users\jbabe\AppData\Roaming\Python\Python310\",

This needs to point to the `fortls.exe`

> "fortran.linter.compilerPath": "C:\Strawbwrry\Fortran\bin\"

and this needs to point to `gfortran.exe`

**Reply by jbaben** [reply]
Hello,
I think the corresponding paths are OK, see the output from the Modern Fortran edition:
[INFO - 12:33:34 PM] Extension Name: Modern Fortran
[INFO - 12:33:34 PM] Extension Version: 3.4.0
[INFO - 12:33:34 PM] Linter set to: "gfortran"
[INFO - 12:33:34 PM] Formatter set to: "fprettify"
[INFO - 12:33:34 PM] Autocomplete set to: "fortls"
[INFO - 12:33:34 PM] Hover set to: "fortls"
[INFO - 12:33:34 PM] Symbols set to: "fortls"
[INFO - 12:33:34 PM] [lint] Found GNU Fortran version 13.2.0

Kind Regards
Juergen

**Comment by jbaben** [comment]
Hello,
When I open the folder with the corresponding FORTRAN code, I get the following messages under Output:
[proc] Befehl wird ausgeführt: cmake --version
[proc] Befehl wird ausgeführt: cmake -E capabilities
[variant] Neue Varianten wurden geladen.
[kit] Erfolgreich geladen: 12 Kits von C:\Users\jbabe\AppData\Local\CMakeTools\cmake-tools-kits.json
[proc] Befehl wird ausgeführt: "C:\Program Files\CMake\bin\cmake.EXE" --version
[proc] Befehl wird ausgeführt: "C:\Program Files\CMake\bin\cmake.EXE" -E capabilities
[proc] Befehl wird ausgeführt: c:\strawberry\Fortran\bin\gcc.exe -v
[proc] Fehler beim Befehl ninja --version. Fehler: Error: spawn ninja ENOENT
[proc] Fehler beim Befehl ninja-build --version. Fehler: Error: spawn ninja-build ENOENT
[proc] Befehl wird ausgeführt: "C:\Program Files\CMake\bin\cmake.EXE" -SE:/MeineProgramme/Visual_Studio_Code/Fortran/Hello -Be:/MeineProgramme/Visual_Studio_Code/Fortran/Hello/build -G "Unix Makefiles"
[proc] Der Befehl ""C:\Program Files\CMake\bin\cmake.EXE" -SE:/MeineProgramme/Visual_Studio_Code/Fortran/Hello -Be:/MeineProgramme/Visual_Studio_Code/Fortran/Hello/build -G "Unix Makefiles"" wurde mit dem Code "1" beendet.
[main] Projekt konfigurieren: Hello 
[proc] Befehl wird ausgeführt: "C:\Program Files\CMake\bin\cmake.EXE" --no-warn-unused-cli -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_C_COMPILER:FILEPATH=c:\strawberry\Fortran\bin\gcc.exe -DCMAKE_CXX_COMPILER:FILEPATH=c:\strawberry\Fortran\bin\g++.exe -SE:/MeineProgramme/Visual_Studio_Code/Fortran/Hello -Be:/MeineProgramme/Visual_Studio_Code/Fortran/Hello/build -G "Unix Makefiles"
[cmake] Not searching for unused variables given on the command line.
[cmake] -- Configuring incomplete, errors occurred!
[cmake] CMake Error at CMakeLists.txt:1:
[cmake]   Parse error.  Expected a command name, got unquoted argument with text
[cmake]   ".SUFFIXES:".
[cmake] 
[cmake] 
[proc] Der Befehl ""C:\Program Files\CMake\bin\cmake.EXE" --no-warn-unused-cli -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_C_COMPILER:FILEPATH=c:\strawberry\Fortran\bin\gcc.exe -DCMAKE_CXX_COMPILER:FILEPATH=c:\strawberry\Fortran\bin\g++.exe -SE:/MeineProgramme/Visual_Studio_Code/Fortran/Hello -Be:/MeineProgramme/Visual_Studio_Code/Fortran/Hello/build -G "Unix Makefiles"" wurde mit dem Code "1" beendet.
My "CMakeLists.txt":
`
.SUFFIXES:
FC=gfortran
COMPILE.f08 = $(FC) $(FCFLAGS) $(TARGET_ARCH) -c

SOURCES=hello.f90

main: $(subst .f90,.o,$(SOURCES))
    $(FC) -o $@ $+

.PHONY: clean
clean:
    -rm -f *.o *.mod *.smod main
`
Kind Regards
Juergen

**Comment by jbaben** [comment]
Sorry,
here the message in english:
[proc] Executing command: cmake --version
[proc] Executing command: cmake -E capabilities
[variant] Loaded new set of variants
[kit] Successfully loaded 12 kits from C:\Users\jbabe\AppData\Local\CMakeTools\cmake-tools-kits.json
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" --version
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" -E capabilities
[proc] Executing command: c:\strawberry\Fortran\bin\gcc.exe -v
[proc] The command: ninja --version failed with error: Error: spawn ninja ENOENT
[proc] The command: ninja-build --version failed with error: Error: spawn ninja-build ENOENT
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" -SE:/MeineProgramme/Visual_Studio_Code/Fortran/Hello -Be:/MeineProgramme/Visual_Studio_Code/Fortran/Hello/build -G "Unix Makefiles"
[proc] The command: "C:\Program Files\CMake\bin\cmake.EXE" -SE:/MeineProgramme/Visual_Studio_Code/Fortran/Hello -Be:/MeineProgramme/Visual_Studio_Code/Fortran/Hello/build -G "Unix Makefiles" exited with code: 1
[main] Configuring project: Hello 
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" --no-warn-unused-cli -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_C_COMPILER:FILEPATH=c:\strawberry\Fortran\bin\gcc.exe -DCMAKE_CXX_COMPILER:FILEPATH=c:\strawberry\Fortran\bin\g++.exe -SE:/MeineProgramme/Visual_Studio_Code/Fortran/Hello -Be:/MeineProgramme/Visual_Studio_Code/Fortran/Hello/build -G "Unix Makefiles"
[cmake] Not searching for unused variables given on the command line.
[cmake] -- Configuring incomplete, errors occurred!
[cmake] CMake Error at CMakeLists.txt:1:
[cmake]   Parse error.  Expected a command name, got unquoted argument with text
[cmake]   ".SUFFIXES:".
[cmake] 
[cmake] 
[proc] The command: "C:\Program Files\CMake\bin\cmake.EXE" --no-warn-unused-cli -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_C_COMPILER:FILEPATH=c:\strawberry\Fortran\bin\gcc.exe -DCMAKE_CXX_COMPILER:FILEPATH=c:\strawberry\Fortran\bin\g++.exe -SE:/MeineProgramme/Visual_Studio_Code/Fortran/Hello -Be:/MeineProgramme/Visual_Studio_Code/Fortran/Hello/build -G "Unix Makefiles" exited with code: 1

Kind Regards
Juergen

**Comment by jbaben** [comment]
Hello,
my solution: I now use: Visual Studio 2022 with the Intel Fortran compiler ifort/ifx.
This allows me to create Fortran source files and create the program.
Kind Regards
Juergen
